src/ja/common/job.py

```diff
@@ -0,0 +1,103 @@
+'''
+This module contains the definition of the Job class and related structures.
+'''
+from enum import Enum
+from docker_context import DockerConstraints, IDockerContext
+
+
+class JobStatus(Enum):
+    '''
+    Represents the status of a job.
+    '''
+    NEW = 0
+    QUEUED = 1
+    RUNNING = 2
+    PAUSED = 3
+    DONE = 4
+    KILLED = 5
+    CRASHED = 6
+
+
+class JobPriority(Enum):
+    '''
+    Represents the priority of a job.
+    '''
+    LOW = 0
+    MEDIUM = 1
+    HIGH = 2
+    URGENT = 3
+
+
+class Job:
+    '''
+    Represents a job with all of its attributes.
+    '''
+    def __init__(self,
+                 owner_id: int,
```

Comment:
I think owner_id should not be in the constructor.
The way I think our project will work:

1. User constructs Job object on user client.
2. Job object is turned into YAML.
3. YAML is transferred to central server.
4. Job object is reconstructed from YAML.
5. Central server validates Job object and **modifies it**.

I think one of the modifications in step 5 will be to set the owner_id.
As such I think that the Job constructor should not contain it because it will only be assigned after object creation.

Reply:
I was thinking that the job is created with the userid by the user client and then the server just checks whether the user communicating with the server is the same one as the user in the job.

Reply:
You're right.
That would work just as well.